src/ui/preferences/MavlinkSettings.qml

```diff
@@ -153,6 +153,38 @@ Rectangle {
                             QGroundControl.isVersionCheckEnabled = checked
                         }
                     }
+
+                    FactCheckBox {
+                        id:         mavlinkForwardingChecked
+                        text:       qsTr("Enable MAVLink forwarding")
+                        fact:       QGroundControl.settingsManager.appSettings.forwardMavlink
+                        visible:    QGroundControl.settingsManager.appSettings.forwardMavlink.visible
+                    }
+
+                    Row {
+                        spacing:    ScreenTools.defaultFontPixelWidth
+                        QGCLabel {
+                            width:              _labelWidth
+                            anchors.baseline:   mavlinkForwardingHostNameField.baseline
+                            visible:            QGroundControl.settingsManager.appSettings.forwardMavlink.rawValue &&
+                                                QGroundControl.settingsManager.appSettings.forwardMavlink.visible
```

Comment:
Should be `QGroundControl.settingsManager.appSettings. QGroundControl.settingsManager.appSettings.forwardMavlinkHostName.visible`. Same thing on the FactTextField below.

Reply:
Also most settings which are of the form enabled setting + value setting disable the secondary setting ui when the enable part is not enabled as opposed to hiding. This provides better usability to the user to understand what the setting is for and what will happen if they enable. As opposed to magic extra stuff that shows up when they check something.

Reply:
Alright I updated this. I made the host name text field always visible but only enabled dependent on the value of the forwardMavlink setting. The visible fields for all the new items are just dependent on the visible property of the forwardMavlink setting, such that if the forwardMavlink setting is disabled everything will go away.

Reply:
"The visible fields for all the new items are just dependent on the visible property of the forwardMavlink setting"

That isn't correct., Each field visibility should be based on the visible property for the specific fact it represents.

Reply:
fixed